src/main/java/com/redhat/devtools/intellij/knative/actions/func/InvokeAction.java

```diff
@@ -96,6 +87,60 @@ public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object s
         }
     }
 
+    private void doInvokeWithRetry(Project project, Kn knCli, InvokeModel model, Function function) throws IOException {
+        String name = function.getName();
+        try {
+            doInvoke(knCli, model, name);
+        } catch (IOException e) {
+            if (e.getLocalizedMessage().contains("function not running")) {
+                int response = UIHelper.executeInUI(() -> Messages.showYesNoDialog(
+                        project,
+                        "Cannot invoke a function if it is not running.\nRun function " + name + "?",
+                        "Unable to invoke func " + name,
+                        "Run",
+                        "Cancel",
+                        AllIcons.General.QuestionDialog
+                ));
+                if (response == Messages.YES) {
+                    RunFuncActionPipeline pipeline = UIHelper.executeInUI(() -> RunAction.Run(project, function, knCli, name));
+                    ((RunFuncActionTask)pipeline.getRunningStep()).setCallbackWhenListeningReady(() -> {
```

Comment:
It looks to me that the callback is set after the process is started so if the runtime and environment is fast, you may miss the message and event ?